arkouda/categorical.py

```diff
@@ -205,7 +206,32 @@ def _binop(self, other : Union[Categorical,str_scalars], op : str_scalars) -> pd
             raise ValueError("Categorical {}: size mismatch {} {}".\
                              format(op, self.size, cast(Categorical,other).size))
         if isinstance(other, Categorical):
-            return self.codes._binop(other.codes, op)
+            if (self.categories.size == other.categories.size) and (self.categories == other.categories).all():
```

Comment:
Is it worth it to add `self.categories.objtype == other.categories.objtype` as an additional clause either in first or second position (short circuit) to weed out `str` vs `pdarray` cases?  (Not sure if that's possible or not but categories looks like it could be either Strings or pdarray.)

Reply:
@glitch good thought, but Categoricals can only be composed of Strings:

![image](https://user-images.githubusercontent.com/10785153/117981247-cd43a100-b302-11eb-87a3-cb50f9c40685.png)

Reply:
ah, gotcha, I see how it's wound together now.